### ISSUE TITLE
False positive unexpected-keyword-arg for user keyword argument in subprocess.Popen()

### ISSUE THREAD

**Issue by bdrung** [PyCQA/pylint#7092]
### Bug description

```python
#!/usr/bin/python3
# pylint: disable=missing-docstring
import os
import subprocess

with subprocess.Popen(["id"], user=os.getlogin()):
    pass
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:6:5: E1123: Unexpected keyword argument 'user' in constructor call (unexpected-keyword-arg)
```

### Expected behavior

Expected no unexpected-keyword-arg error.

### Pylint version

```shell
pylint 2.14.4
astroid 2.11.6
Python 3.10.5 (main, Jun  8 2022, 09:26:22) [GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04 (jammy)
